### ISSUE TITLE
Not working twig (`||` instead of `or`) is not an error

### ISSUE THREAD

**Issue by aragon999** [MalteJanz/ludtwig#43]
Thank you for the linter :-)

If you have some twig statement like `{% if foo > 0 || bar < 0 %}` it is actually not valid and only works at some specific version of Shopware, where it has been implemented as alias for `or`.

I think this should be seen as an error.

The same is probably true for `and` and `&&`.

**Comment by MalteJanz** [comment]
Thank you for the bug report 😄 ,

unfortunately this hobby project is almost dead because we now use ESLint (with proper Vue.js linting) at shopware for the administration (which is far more powerful for our use case). I actually don't know what is used for the storefront currently (I am studying at the moment 👨‍🎓 ).

This project mostly started as a hobby to learn / use the Rust language and improve the workflow at shopware a bit but never gained any real traction. It is also mostly a formatter and doesn't care much about the syntax inside the twig if statement yet (you could write anything after the if and before the closing bracket if I remember correctly - this was done to get parsing to a basic abstract syntax tree working and apply a uniform formatting across the shopware adminstration and storefront).

feel free to fork the project and make the changes necessary to report errors on encountering `||` / `&&` if you want. It shouldn't be too hard if you know some basic Rust and look inside the `analyzer.rs` file (you could simply check the node type there for a twig if and check the condition string without touching all the parsing logic in my other repository). Something like this added here https://github.com/MalteJanz/ludtwig/blob/6a747ec2a486e2188a29a5f87ced3d275af3513d/src/analyzer.rs#L27 :
```rust
if let SyntaxNode::TwigStructure(TwigStructure::TwigIf(twig_if)) = node {
            for if_arm in &twig_if.if_arms {
                if let Some(e) = &if_arm.expression {
                    if e.contains("&&") {
                        file_context.send_output(Output::Warning(
                            "Invalid twig if expression syntax found: '&&'".to_owned(),
                        ));
                        continue;
                    }

                    if e.contains("||") {
                        file_context.send_output(Output::Warning(
                            "Invalid twig if expression syntax found: '||'".to_owned(),
                        ));
                        continue;
                    }
                }
            }
        }
```
But unfortunately the analyzer isn't capable to produce line numbers in its output because of a design decision flaw (which I never refactored - both the analyzing logic and the formatted printing happen in parallel).

I hope this answer helps you somehow 🙂.

**Comment by MalteJanz** [comment]
Fixed in #44
